lib/teleterm/apiserver/handler/handler_access_requests.go

```diff
@@ -0,0 +1,155 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package handler
+
+import (
+	"context"
+
+	api "github.com/gravitational/teleport/lib/teleterm/api/protogen/golang/v1"
+	"github.com/gravitational/teleport/lib/teleterm/clusters"
+
+	"github.com/gravitational/trace"
+)
+
+func (s *Handler) GetRequestableRoles(ctx context.Context, req *api.GetRequestableRolesRequest) (*api.GetRequestableRolesResponse, error) {
+	roles, err := s.DaemonService.GetRequestableRoles(ctx, req)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	return roles, nil
+}
+
+// GetAccessRequests returns a list of all available access requests the user can view
+func (s *Handler) GetAccessRequests(ctx context.Context, req *api.GetAccessRequestsRequest) (*api.GetAccessRequestsResponse, error) {
+	requests, err := s.DaemonService.GetAccessRequests(ctx, req)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	response := &api.GetAccessRequestsResponse{}
+	for _, req := range requests {
+		response.Requests = append(response.Requests, newApiAccessRequest(req))
+	}
+
+	return response, nil
+}
+
+// GetAccessRequest returns a single access request by id
+func (s *Handler) GetAccessRequest(ctx context.Context, req *api.GetAccessRequestsRequest) (*api.GetAccessRequestResponse, error) {
+	requests, err := s.DaemonService.GetAccessRequests(ctx, req)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	response := &api.GetAccessRequestResponse{}
+	if len(requests) < 1 {
+		return nil, trace.NotFound("access request %q not found", req.Id)
+	}
+	response.Request = newApiAccessRequest(requests[0])
+
+	return response, nil
+}
+
+func (s *Handler) CreateAccessRequest(ctx context.Context, req *api.CreateAccessRequestRequest) (*api.CreateAccessRequestResponse, error) {
+	request, err := s.DaemonService.CreateAccessRequest(ctx, req)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	createdRequest := &api.CreateAccessRequestResponse{
+		Request: newApiAccessRequest(*request),
+	}
+	return createdRequest, nil
+}
+
+func (s *Handler) DeleteAccessRequest(ctx context.Context, req *api.DeleteAccessRequestRequest) (*api.DeleteAccessRequestResponse, error) {
+	err := s.DaemonService.DeleteAccessRequest(ctx, req)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return &api.DeleteAccessRequestResponse{
+		Success: "true",
+	}, nil
+}
+
+func (s *Handler) AssumeRole(ctx context.Context, req *api.AssumeRoleRequest) (*api.AssumeRoleResponse, error) {
+	err := s.DaemonService.AssumeRole(ctx, req)
```

Comment:
Connect let me create a resource access request and then approve it from another account. However, when I log back in and try to assume it, it fails and simply shows "7 PERMISSION_DENIED: access denied".

When I look at the tshd logs, I see this:

> Original Error: *trace.AccessDeniedError this Teleport cluster is not licensed for resource access requests, please contact the cluster administrator

There are two problems:

1. Surely we should fail sooner than that? Perhaps the UI for resource access requests shouldn't even be shown if the cluster isn't licensed for them.
2. The error message that's shown in the UI is too vague. The error message from the logs is much better but we lose it along the way somehow. I'm not sure what the issue is, perhaps we recreate the error somewhere instead of wrapping it? I haven't investigated that so it might as well be an issue outside of lib/teleterm's code.
 

![access denied](https://user-images.githubusercontent.com/27113/192498557-ce81e1bd-afa1-4012-9a76-8a75eadf9c79.png)

Reply:
I've encountered this error when I 'downgraded' from enterprise to oss. Creating a resource access request requires you to have a 'search_as_roles' value in a role and, this can only be _set_ with enterprise. It's due to the fact that when you downgrade, we don't check all of your roles configurations. This same behavior happens with `tsh` as well. I think this behavior is out of scope of this PR. 
However, once we add a 'feature block' type of thing for Connect with non-enterprise clusters, this should go away as well.

Reply:
Ah, I think it might have been due to the fact that I was reviewing SBARs when they were not yet hidden behind the license file. I had an old license file which didn't have access to new features.